src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/ImporterService.cs

```diff
@@ -181,14 +181,27 @@ await _dataImportService.UpdateStatus(dataImport.Id,
                             .Select(f => f.AutoSelectFilterItemLabel)
                             .SingleOrDefault() ?? "Total"; // If meta file didn't specify a default, look for "Total"
 
-                        return filterItems
+                        var filterItemsWithAutoSelectLabel = filterItems
                             .Where(item =>
-                                item.FilterGroup.FilterId == filterId
-                                && item.Label.Equals(autoSelectFilterItemLabel, OrdinalIgnoreCase))
-                            .Select(item => new { item.Id , item.Label})
-                            // There might be two filter items with the same label under different groups.
-                            // If so, we set no autoSelectFilterItem.
-                            .SingleOrDefault();
+                                item.FilterGroup.FilterId == filterId &&
+                                item.Label.Equals(autoSelectFilterItemLabel, OrdinalIgnoreCase));
+
+                        // There might be two or more filter items with the same label under different groups
+                        if (filterItemsWithAutoSelectLabel.Count() > 1)
+                        {
+                            // If so, does one belong to a filter group with the same label?
+                            var filterItemWithMatchingGroupLabel = filterItemsWithAutoSelectLabel.SingleOrDefault(fi => fi.FilterGroup.Label == autoSelectFilterItemLabel);
+
+                            if (filterItemWithMatchingGroupLabel is not null)
+                            {
+                                // If it does, this is the auto select filter item
+                                return new { filterItemWithMatchingGroupLabel.Id, filterItemWithMatchingGroupLabel.Label };
+                            }
```

Comment:
We need to return _something_ from the `if (filterItemsWithAutoSelectLabel.Count > 1)` block or we'll incorrectly fall through to the `filterItemsWithAutoSelectLabel == 0` default case which is almost certainly a bug.

This can be fixed with something like:

```cs
if (filterItemsWithAutoSelectLabel.Count > 1)
{
    return filterItemsWithAutoSelectLabel
        .Where(fi => fi.FilterGroup.Label == autoSelectFilterItemLabel)
        .Select(fi => new { fi.Id, fi.Label })
        .SingleOrDefault();
}
```

Note this above snippet doesn't take into account the two cases we also need to cover that I've mentioned in the earlier comment above.